lib/ansible/plugins/inventory/ini.py

```diff
@@ -104,15 +121,32 @@ def __init__(self) -> None:
 
         self.patterns: dict[str, re.Pattern] = {}
         self._origin: Origin | None = None
+        self._show_deprecation = False
```

Comment:
It would be more clear to be specific about what this flag represents, since it's possible there could be additional deprecations in the future:

```suggestion
        self._invalid_extension = False
```